src/v/cluster_link/link.cc

```diff
@@ -90,7 +98,16 @@ ss::future<> link::stop() {
           "Stopping task {} for cluster link {}",
           t->name(),
           _config.name);
-        auto res = co_await stop_task(t.get());
+        auto res_f = co_await ss::coroutine::as_future(stop_task(t.get()));
+        if (res_f.failed()) {
+            vlog(
+              cllog.warn,
+              "Failed to stop task {}: {}",
+              t->name(),
+              res_f.get_exception());
```

Comment:
FYI i don't think we should be doing this--correctness here depends on `vlog` always evaluating all of its parameters otherwise we get abanonded exceptional future exception here. it's similar to avoiding ever having side effects in an assert.